pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarRecordCursor.java

```diff
@@ -418,7 +421,16 @@ public boolean advanceNextPosition() {
         //start time for deseralizing record
         metricsTracker.start_RECORD_DESERIALIZE_TIME();
 
-        currentRecord = this.schemaHandler.deserialize(this.currentMessage.getData());
+        if (this.schemaHandler instanceof KeyValueSchemaHandler) {
+            ByteBuf keyByteBuf = null;
+            if (this.currentMessage.getKey().isPresent()) {
+                keyByteBuf = Unpooled.wrappedBuffer(
+                        Base64.getDecoder().decode(this.currentMessage.getKey().get()));
+            }
```

Comment:
We can get the `keyBytes` by  `message.getKeyBytes()`, so that you don't need to decode a String by base64 decoder.